src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -346,36 +359,44 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
       setValue(key, amount)
 
       let estimatedFees
+      let hasEnoughNativeTokenForGas = false
 
       try {
-        estimatedFees = await estimateFormFees()
-        setValue(SendFormFields.EstimatedFees, estimatedFees)
+        // Make API call on first user input
+        // otherwise, debounce user input and use last call
+        if (inputValue.length === 1) {
```

Comment:
This feels a little hacky, and doesn't work for the edge case of pasting values.
We also lose the snappy feeling of a leading edge request when entering a debounce window when values are already entered.

Can we instead do it such that if we aren't waiting on any responses, fire off a request?
I.e. change logic from `inputValue.length === 1` to _if first request_.

Reply:
Yes. The copy and paste is a good case I hadn't tested yet. Will update this to be more generic.

Reply:
Fires now on first input with debounce { leading: true }, which handles pasting in values.